libs/dh/imbalance-prices/shell/src/lib/drawer/dh-drawer.component.ts

```diff
@@ -117,12 +117,27 @@ export class DhImbalancePricesDrawerComponent {
   private readonly toastService = inject(WattToastService);
   private readonly httpClient = inject(ImbalancePricesHttp);
   private readonly apollo = inject(Apollo);
+  private readonly locale = inject(MAT_DATE_LOCALE) as string;
 
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  private readonly year = computed(() => Number(format(this.imbalancePrice()!.name, 'yyyy')));
+  private readonly year = computed(() => {
+    const imbalancePrice = this.imbalancePrice();
 
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  private readonly month = computed(() => Number(format(this.imbalancePrice()!.name, 'M')));
+    if (!imbalancePrice) return 0;
+
+    const formatter = new Intl.DateTimeFormat(this.locale, { year: 'numeric' });
+    const formattedDate = formatter.format(imbalancePrice.name);
+    return Number(formattedDate);
+  });
+
+  private readonly month = computed(() => {
+    const imbalancePrice = this.imbalancePrice();
+
+    if (!imbalancePrice) return 0;
+
+    const formatter = new Intl.DateTimeFormat(this.locale, { month: 'numeric' });
+    const formattedDate = formatter.format(imbalancePrice.name);
+    return Number(formattedDate);
+  });
```

Comment:
suggestion: let's reuse `dayjs` to format dates. It's been used everywhere else anyway.